python/iot3/src/iot3/core/__init__.py

```diff
@@ -0,0 +1,211 @@
+
+    for comp in _otel.Compression:
+        if config["otel"]["compression"] == comp.value:
+            o_kwargs["compression"] = comp
+            break
+    else:
+        raise ValueError(f"unknown compression {config['otel']['compression']}")
+
+    o = _otel.Otel(
+        service_name=config["otel"]["service_name"],
+        endpoint=config["otel"]["endpoint"],
+        batch_period=config["otel"]["batch_period"],
+        max_backlog=config["otel"]["max_backlog"],
+        **o_kwargs,
+    )
+    o.start()
+
+    # Wrap the callback to avoid leaking telemetry into the simple API
+    def _msg_cb(*, data, topic, payload, **_kwargs):
+        message_callback(data, topic, payload)
+
+    m = _mqtt.MqttClient(
+        client_id=config["mqtt"]["client-id"],
+        host=config["mqtt"]["host"],
+        port=config["mqtt"]["port"],
+        username=config["mqtt"]["username"],
+        password=config["mqtt"]["password"],
+        msg_cb=_msg_cb if message_callback else None,
+        msg_cb_data=callback_data,
+        span_ctxmgr_cb=o.span,
+    )
+    m.start()
+
+    _core = dict([("otel", o), ("mqtt", m)])
+
+
+def stop():
+    """Stop the Iot3 Core SDK."""
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    _core["mqtt"].stop()
+    _core["otel"].stop()
+    _core = None
+
+
+def is_ready() -> bool:
+    """Checkxs whether the IoT3 Core SDK is ready.
+
+    The IoT3 Core SDK is ready when the MQTT client is connected to
+    the MQTT broker. Messages sent when the SDK is not ready, are
+    lost without notice.
+    """
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    return _core["mqtt"].is_ready()
+
+
+def wait_for_ready():
+    """Wait until the IoT2 Core SDK is ready.
+
+    Beware that this may take an undeterminate amount of time; in
+    case the MQTT client can't connect at all, wait_for_ready()
+    will wait forever and never return.
+    """
+    while True:
+        if ready():
+            break
+        _time.sleep(0.1)
+
+
+def publish(
+    topic: str,
+    payload: str | bytes,
+):
+    """Send an MQTT message
+
+    "param topic: The MQTT topic on which to send the message.
+    :param payload: The payload of the MQTT message to send.
+    """
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    _core["mqtt"].publish(topic=topic, payload=payload)
+
+
+def subscribe(
+    topics: str | list[str],
+):
+    """Subscribe to a list of MQTT topics.
+
+    If a subscription already existed, it is entirely replaced.
+
+    :param topics: A topic, or a list of topics.
+    """
+    global _core
+
+    if _core is None:
+        raise RuntimeError("IoT3 Core SDK not initialised.")
+
+    topics = topics if isinstance(topics, list) else [topics]
+    _core["mqtt"].subscribe_replace(topics=topics)
+
+
+_core = None
```

Comment:
Could you explain me?

Reply:
This variable holds the context of the simple API. It is accessed as
a `global` in the various functions, so it must be defined at the
module level.